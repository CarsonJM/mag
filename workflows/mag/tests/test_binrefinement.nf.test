nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_binrefinement"

    test("Parameters: refine_bins_dastool = true") {

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                ).match() }
            )
        }
    }
}
