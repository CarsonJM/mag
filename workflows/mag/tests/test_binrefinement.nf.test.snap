{
    "Parameters: refine_bins_dastool = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut.contigs.fa:md5,49bd2dbd085e37b3f19a62d9d010a271"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_contigs.fasta:md5,86027592505360f0e51a395555b8a11b"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut_sample2.contigs.fa:md5,378c9d74964ef6a9a0fd2f2c77dfa607"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_sample2_contigs.fasta:md5,0c3fdd6f534d5e21b105be45351ee655"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "MEGAHIT-MetaBAT2Refined-test_minigut.1.fa:md5,489a5f7ba3795477e09d2d396fe535c3",
                            "MEGAHIT-MetaBAT2Refined-test_minigut.2.fa:md5,1addf634bca82af85d36621585f93f52"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut.001_sub.fa:md5,b1633cb560c41afe3606b6e8f2082211",
                            "SPAdes-MetaBAT2Refined-test_minigut.2.fa:md5,7f8be703a086fd12d41d2df817a08e8d"
                        ]
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "MEGAHIT-DASToolUnbinned-test_minigut.fa:md5,5b68d8b2df39cb4ab46299d9a09f1243"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "SPAdes-DASToolUnbinned-test_minigut.fa:md5,067f6ad8e2f820ab3d6000a154150954"
                    ]
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,3550550b622b6d5f6803ffed9742e08b",
                    "versions.yml:md5,586dc48e8e2cc8106cb0ee3b70ce0fef",
                    "versions.yml:md5,6d41fb977d31ce96d7ddbb4ea2dbe38b",
                    "versions.yml:md5,7017c8959eb098440346ca9a2310c460",
                    "versions.yml:md5,7ccffaba4e75be98b8faedb54ccea16f",
                    "versions.yml:md5,8187d1683f97b64a582620caba2b175a",
                    "versions.yml:md5,82c86a3130605a0de153c903b2cfc6d2",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,aee39cb21305f6d3fe88bf9863529706",
                    "versions.yml:md5,b4ee8f08c1efaac6dee44f113c7a2e0b",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,c384e0cacaa344670dcde728d78a524a",
                    "versions.yml:md5,c777f00c6a48f3c0490d09b1aa8d81a6",
                    "versions.yml:md5,cc4a4972327dec07faf1f54ead653715",
                    "versions.yml:md5,d8d2321bec0c70de2f9750f258fc890d",
                    "versions.yml:md5,dc29f891cb297131cf940ba5580a6832",
                    "versions.yml:md5,df654dd4fdb4039c2d8cf0b6b49f4b55",
                    "versions.yml:md5,e9f8fe610b5abe68b28d77efc951e9e6",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c"
                ],
                "assemblies": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut.contigs.fa:md5,49bd2dbd085e37b3f19a62d9d010a271"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_contigs.fasta:md5,86027592505360f0e51a395555b8a11b"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut_sample2.contigs.fa:md5,378c9d74964ef6a9a0fd2f2c77dfa607"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_sample2_contigs.fasta:md5,0c3fdd6f534d5e21b105be45351ee655"
                    ]
                ],
                "bin_summary": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "MEGAHIT-MetaBAT2Refined-test_minigut.1.fa:md5,489a5f7ba3795477e09d2d396fe535c3",
                            "MEGAHIT-MetaBAT2Refined-test_minigut.2.fa:md5,1addf634bca82af85d36621585f93f52"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut.001_sub.fa:md5,b1633cb560c41afe3606b6e8f2082211",
                            "SPAdes-MetaBAT2Refined-test_minigut.2.fa:md5,7f8be703a086fd12d41d2df817a08e8d"
                        ]
                    ]
                ],
                "refined_unbins": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "MEGAHIT-DASToolUnbinned-test_minigut.fa:md5,5b68d8b2df39cb4ab46299d9a09f1243"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "SPAdes-DASToolUnbinned-test_minigut.fa:md5,067f6ad8e2f820ab3d6000a154150954"
                    ]
                ],
                "short_reads": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,3550550b622b6d5f6803ffed9742e08b",
                    "versions.yml:md5,586dc48e8e2cc8106cb0ee3b70ce0fef",
                    "versions.yml:md5,6d41fb977d31ce96d7ddbb4ea2dbe38b",
                    "versions.yml:md5,7017c8959eb098440346ca9a2310c460",
                    "versions.yml:md5,7ccffaba4e75be98b8faedb54ccea16f",
                    "versions.yml:md5,8187d1683f97b64a582620caba2b175a",
                    "versions.yml:md5,82c86a3130605a0de153c903b2cfc6d2",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,aee39cb21305f6d3fe88bf9863529706",
                    "versions.yml:md5,b4ee8f08c1efaac6dee44f113c7a2e0b",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,c384e0cacaa344670dcde728d78a524a",
                    "versions.yml:md5,c777f00c6a48f3c0490d09b1aa8d81a6",
                    "versions.yml:md5,cc4a4972327dec07faf1f54ead653715",
                    "versions.yml:md5,d8d2321bec0c70de2f9750f258fc890d",
                    "versions.yml:md5,dc29f891cb297131cf940ba5580a6832",
                    "versions.yml:md5,df654dd4fdb4039c2d8cf0b6b49f4b55",
                    "versions.yml:md5,e9f8fe610b5abe68b28d77efc951e9e6",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c",
                    "versions.yml:md5,ede79d94ca30c948c1707584760b230c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-08T14:25:50.952306304"
    }
}